tests/hacbs-demo/demo.go

```diff
@@ -0,0 +1,109 @@
+package build
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/devfile/library/pkg/util"
+	"github.com/google/uuid"
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"k8s.io/klog"
+	"knative.dev/pkg/apis"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+)
+
+const (
+	helloWorldComponentGitSourceRepoName = "devfile-sample-hello-world"
+	pythonComponentGitSourceURL          = "https://github.com/redhat-appstudio-qe/devfile-sample-python-basic"
+)
+
+var (
+	helloWorldComponentGitSourceURL = fmt.Sprintf("https://github.com/%s/%s", utils.GetEnv("GITHUB_E2E_ORGANIZATION", "redhat-appstudio-qe"), helloWorldComponentGitSourceRepoName)
+)
+
+var _ = framework.BuildSuiteDescribe("HACBS demo", Label("hacbs", "hacbs-demo"), func() {
+	defer GinkgoRecover()
+
+	f, err := framework.NewFramework()
+	Expect(err).NotTo(HaveOccurred())
+
+	var applicationName, componentName, testNamespace, outputContainerImage string
+	var timeout, interval time.Duration
+
+	Describe("HACBS-683 Build a component using HACBS pipeline", Ordered, func() {
+		BeforeAll(func() {
+			testNamespace = utils.GetEnv(constants.E2E_APPLICATIONS_NAMESPACE_ENV, fmt.Sprintf("hacbs-e2e-%s", util.GenerateRandomString(4)))
+			_, err := f.HacbsUser.CommonController.CreateTestNamespace(testNamespace)
+			Expect(err).NotTo(HaveOccurred(), "Error when creating/updating '%s' namespace: %v", testNamespace, err)
+
+			_, err = f.HacbsUser.CommonController.CreateRegistryAuthSecret(
+				"redhat-appstudio-registry-pull-secret",
```

Comment:
Do we will need to create this secret all time when we push components?

Reply:
yes, in kcp world the shared secret feature doesn't work (at least not now afaik) - users would have to provide their own secrets. back to the past, gentlemen :)